pkg/apis/ark/v1/status_request.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2018 the Heptio Ark contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// StatusRequest is a request to access current status information about
+// the Ark server.
+type StatusRequest struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	Spec   StatusRequestSpec   `json:"spec"`
+	Status StatusRequestStatus `json:"status,omitempty"`
+}
+
+// StatusRequestSpec is the specification for a StatusRequest.
+type StatusRequestSpec struct {
+}
+
+// StatusRequestPhase represents the lifecycle phase of a StatusRequest.
+type StatusRequestPhase string
+
+const (
+	// StatusRequestPhaseNew means the StatusRequest has not been processed
+	// by the StatusRequestController yet.
+	StatusRequestPhaseNew StatusRequestPhase = "New"
+	// StatusRequestPhaseProcessed means the StatusRequest has been processed
+	// by the StatusRequestController.
+	StatusRequestPhaseProcessed StatusRequestPhase = "Processed"
+)
+
+// StatusRequestStatus is the current status of a StatusRequest.
+type StatusRequestStatus struct {
+	// Phase is the current lifecycle phase of the StatusRequest.
+	Phase StatusRequestPhase `json:"phase"`
+
+	// ProcessedTimestamp is when the StatusRequest was processed
+	// by the StatusRequestController.
```

Comment:
I would omit the controller detail - that's not something an end user would want or need to know about.